lib/generators/rspec/controller/controller_generator.rb

```diff
@@ -15,8 +17,9 @@ class ControllerGenerator < Base
       def generate_request_spec
         return unless options[:request_specs]
 
-        template 'request_spec.rb',
-                 File.join('spec/requests', class_path, "#{file_name}_request_spec.rb")
+        template_file = actions.empty? ? 'request_spec.rb' : 'request_with_actions_spec.rb'
```

Comment:
You've renamed the file so where does the alternate come from? Any differences are also untested which I'd like to see rectified!

Reply:
> You've renamed the file so where does the alternate come from?

The alternate template comes from the integration generator (added on the source_paths on line 7).

> Any differences are also untested which I'd like to see rectified!

It is reusing functionality of requests generator (and integration generator) because it is just re-using the template, but it is actually being tested on the [shared example](https://github.com/rspec/rspec-rails/pull/2356/commits/6cffc0549b6a7fa25ab2421f23c20d4da4c3daa0#diff-09c325a5eb95b7e1e766904a00127223R7).

Testing with that shared example we make sure that generators that generate request specs does use a consistent format (they all pass the same specs) while differences between specs are being also being tested.

Reply:
Maybe generate `posts` with a scaffold, and generate `books` with controller generator?

Reply:
You haven't added a test for the different behaviour, and there is different behaviour now.

Reply:
@JonRowe I did add a shared example that it's testing just that new behavior. You can check that removing that code and specs will fail with meaningful errors.

The specs for that code are in [this support file](https://github.com/rspec/rspec-rails/blob/master/spec/support/generators.rb#L40). I'm just adding one line of test because I'm reusing existing specs.

Does it makes sense?

Reply:
@JonRowe Runing the tests without the implementation changes gives those [rspec failures](https://gist.github.com/eloyesp/196cae8f0c36598609479d4c71441e70).